controllers/provisionserver_controller.go

```diff
@@ -0,0 +1,310 @@
+// +kubebuilder:rbac:groups=security.openshift.io,namespace=openstack,resources="securitycontextconstraints",resourceNames="privileged",verbs="use"
+// +kubebuilder:rbac:groups=security.openshift.io,namespace=openstack,resources="securitycontextconstraints",resourceNames="anyuid",verbs="use"
+
+// Reconcile - provision image servers
+func (r *ProvisionServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	_ = r.Log.WithValues("provisionserver", req.NamespacedName)
+
+	// Fetch the ProvisionServer instance
+	instance := &ospdirectorv1beta1.ProvisionServer{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers. Return and don't requeue.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// config maps
+	op, err := r.httpdConfigMapCreateOrUpdate(instance)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("Httpd ConfigMap %s successfully reconciled - operation: %s", instance.Name, string(op)))
+	}
+
+	// provisionserver
+	// Create or update the Deployment object
+	op, err = r.deploymentCreateOrUpdate(instance)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
+	}
+
+	// Get the provisioning IP of node hosting the pod belonging to the CR
+	podIP, err := r.getProvisionServerProvisioningIP(instance)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if podIP == "" {
+		r.Log.Info(fmt.Sprintf("Deployment %s pod provisioning IP not yet available, requeuing and waiting", instance.Name))
+		return ctrl.Result{RequeueAfter: time.Second * 30}, err
+	}
+
+	// Update Status
+	instance.Status.LocalImageURL = r.getLocalImageURL(podIP, instance)
+	err = r.Client.Status().Update(context.TODO(), instance)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CR status %v")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager - prepare controller for use with operator manager
+func (r *ProvisionServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// TODO: Myabe use filtering functions here since some resource permissions
+	// are now cluster-scoped?
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ospdirectorv1beta1.ProvisionServer{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&corev1.ConfigMap{}).
+		Complete(r)
+}
+
+func (r *ProvisionServerReconciler) httpdConfigMapCreateOrUpdate(instance *ospdirectorv1beta1.ProvisionServer) (controllerutil.OperationResult, error) {
```

Comment:
see above configmap creation from template comment. this would remove the req for this